qmi/instruments/newport/smc_100cc.py

```diff
@@ -30,3 +44,542 @@ def __init__(self,
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
+            "KD", derivative_gain))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        if persist:
+            self.exit_configuration_state(controller_address)
+
+    @rpc_method
+    def get_integral_gain(self, controller_address: Optional[int] = None) -> float:
+        """
+        Get the integral gain of the PID control loop.
+
+        Parameters:
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised value of the controller address.
+
+        Returns:
+            The integral gain of the PID control loop.
+        """
+        _logger.info(
+            "Getting integral gain of the PID control loop of instrument [%s]", self._name)
+        # instrument must be in configuration state to get the integral gain of the PID control loop value.
+        self.reset(controller_address)
+        self.enter_configuration_state(controller_address)
+        res = self._scpi_protocol.ask(
+            self._build_command("KI?"))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        self.exit_configuration_state(controller_address)
+        return float(res[3:])
+
+    @rpc_method
+    def set_integral_gain(
+            self, integral_gain: float, persist: bool = False, controller_address: Optional[int] = None) -> None:
+        """
+        Set the integral gain of the PID control loop.
+
+        Parameters:
+            integral_gain:    New integral_gain value in Volt * second/preset unit.
+            persist:            Flag to indicate if the integral gain should be persisted to the controller's memory,
+                                so it is still available after powering down the controller. When not persisted, the
+                                integral gain is the one stored in the controller's memory.
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised integral_gain of the controller address.
+        """
+        if 1E12 <= integral_gain or integral_gain < 0:
+            raise QMI_InstrumentException(
+                f"Provided value {integral_gain} not in valid range 0 >= integral_gain > 1E12.")
+
+        _logger.info(
+            "Setting integral gain of the PID control loop of instrument [%s] to [%f]",
+            self._name, integral_gain
+        )
+        self.controller_address = controller_address
+        # instrument must be in configuration state to persist the set integral gain of the PID control loop.
+        if persist:
+            self.reset(controller_address)
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
```

Comment:
Same ass the `FD` command. It cannot be called in the Ready state. So you would have to make sure the controller is in the Disable state here